web-api/openapi/front_office.yaml

```diff
@@ -152,6 +159,21 @@ components:
       description: Include the properties of each document in the response
       type: boolean
       default: true
+    FilterString:
+      type: object
+      additionalProperties:
+        type: object
+        properties:
+          $eq:
+            $ref: ./schemas/document.yml#/DocumentPropertyString
+        minProperties: 1
+        maxProperties: 1
```

Comment:
with `FilterEq` you skip one level of nesting. basically we want compare filter queries of the form
```
{ <property_name>: { <compare_op>: <property_value>    } }
|<FilterString>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>|
                   |<FilterString.additionalProperties>|
```

Reply:
What I'm saying is:

```
{ <property_id>: { <compare_op>: <property_value>             } }
|<LeafFilter object with additionalProperties>>>>>>>>>>>>>>>>>>>|
                 |<FilterEq object>>>>>>>>>>>>>>>>>>>>>>>>>>>|
```

Reply:
Because we don't only have string filter and the object  which contains the property id shouldn't be specific to any type of the filter operation it contains.

Reply:
if i'm not mistaken this is already facilitated by renaming the `FilterString` to `FilterCompare` (currently its inner object only has one property, but that's equivalent to `oneOf` with a single item anyways). so for example the current definition is
```yaml
FilterCompare:
  type: object
  additionalProperties:
    type: object
    properties:
      $eq:
        $ref: ./schemas/document.yml#/DocumentPropertyString
    minProperties: 1
    maxProperties: 1
```
and in the following prs it will be extended to for example
```yaml
FilterCompare:
  type: object
  additionalProperties:
    type: object
    properties:
      $eq:
        oneOf:
          - $ref: ./schemas/document.yml#/DocumentPropertyString
          - $ref: ./schemas/document.yml#/DocumentPropertyBool
      $in:
        $ref: ./schemas/document.yml#/DocumentPropertyArrayString
    minProperties: 1
    maxProperties: 1
```
this is essentially the same, just without naming the inner object.